### PR TITLE
New GSIM Idini2017

### PR DESCRIPTION
**Pull request by pheresi**
- Implements Idini et al (2017) gsim, including its test and verification tables
- Includes soiltype into site.py and adds it in site_test.py and oqvalidation.py

### CODE REVIEW

**Comment by micheles** [comment]
Since you are extending the site model we also need an end-to-end test using a site model with `soiltype`. You can copy what has been done in openquake/qa_tests_data/classical/case_37 for the siteclass parameter.

**Comment by pheresi** [comment]
Oh ok, perfect. Should I create a new case (case_38) from case_37, and include it in this pull request, or create another one?

**Comment by pheresi** [comment]
Oops... just saw that there is already a case_38... So well, I mean case_63

**Comment by micheles** [comment]
You are probably doing more work than needed. Adding `soiltype` should be a single line. See for instance this PR that is adding
the site parameter `bas`: https://github.com/gem/oq-engine/pull/6672/files

**Comment by pheresi** [comment]
Sorry, sorry... yesterday I couldn't work on this, but I'll do it today.

**Comment by micheles** [comment]
There are some conflicts to resolve

**Comment by pheresi** [comment]
Oh, right. I'm going to resolve those in a bit. Thanks for the reminder!

**Comment by pheresi** [comment]
Hi again,
The engine check failed in case_13 of classical PSHA. Should I just add "soiltype" to the .csv file where it failed?
Also, the hazardlib check failed while installing dependencies, but I'm not sure why. Can you help me with this one?
Thanks!

File: openquake/hazardlib/site.py [hunk 1]
```diff
@@ -252,6 +253,7 @@ def from_points(cls, lons, lats, depths=None, sitemodel=None,
             self._set('z2pt5', sitemodel.reference_depth_to_2pt5km_per_sec)
             self._set('siteclass', sitemodel.reference_siteclass)
             self._set('backarc', sitemodel.reference_backarc)
+            self._set('soiltype', sitemodel.reference_soiltype)
```
Comment (micheles) [inline openquake/hazardlib/site.py:256]: Please remove this line and the test should become green again

Comment (pheresi) [reply openquake/hazardlib/site.py:256]: But then how can we define a reference soiltype for all the sites in a region?

Comment (micheles) [reply openquake/hazardlib/site.py:256]: Just write a site model with soiltype constant for all sites. If you look at the PR that I mentioned before (https://github.com/gem/oq-engine/pull/6672/files) the "bas" term is introduced only as a local site parameter. Not introducing `reference_soiltype` not only makes your life easier, it also avoid building a site collection larger than needed for all models not using Idini2017, i.e. all model in existence before this PR ;-)
It was a mistake to introduce the siteclass as a global parameter, I should remove that one too.

Comment (pheresi) [reply openquake/hazardlib/site.py:256]: oh, you are totally right! let me remove the line update

**Comment by micheles** [comment]
You forgot to commit case_63/expected/hcurve-mean.csv

**Comment by pheresi** [comment]
Yep, was a mistake with the file names... updating! :)

**Comment by pheresi** [comment]
Umm... Im not sure why the classical_test failed in my case_63

**Comment by micheles** [comment]
It looks looks like a wrong newline ending. And now there are conflicts again, sorry ;-)

**Comment by pheresi** [comment]
Got it! Updating :)

**Comment by pheresi** [comment]
Finally! :)

Michele, thanks for your help, and sorry for the spam!

**Review by mmpagani — APPROVED** [review]
lgtm

**Pull request merged**